File: Balloon Everywhere.java
class Solution {
    public int maxInstance(String s) {
        if(s.length() < 7) return 0;
        int arr[] = new int[26];
        for(char ch : s.toCharArray())
        arr[ch-'a']++;
        int singleOccuring = Math.min(arr['b'-'a'], Math.min(arr[0], arr['n'-'a']));
        int doubleOccuring = Math.min(arr['l'-'a'], arr['o'-'a']);
        doubleOccuring /= 2;
        return Math.min(doubleOccuring , singleOccuring);
    }
}

File: Balloon Everywhere.cpp
class Solution{

public:

    int maxInstance(string s){

        vector<int> m(27,0) ;

        int i = 0 ;

        while(s[i]!='\0')

        {

            if(s[i]=='b' || s[i]=='a' || s[i]=='l' || s[i]=='o' || s[i]=='n')

            {

                m[s[i]-96]++ ;

            }

            i++ ;

        }

        int ans = m['b'-96]/1 ;

        ans = min(ans , m['a'-96]/1) ;

        ans = min(ans , m['l'-96]/2) ;

        ans = min(ans , m['o'-96]/2) ;

        ans = min(ans , m['n'-96]/1) ;

        return ans ;

    }

 

};
